### PR TITLE
[Test] Use a flag to separate integration test from sample test

### PR DESCRIPTION
**Pull request by numerology**
Part of #3256

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3298)

### CODE REVIEW

File: test/sample_test.yaml [hunk 1]
```diff
@@ -63,29 +66,46 @@ spec:
...
               - volume_snapshot_ops
-              - kubeflow_tf_serving
-              - container_build
               - loop_output
               - loop_parameter
               - loop_static
               - resource_ops
               - multiple_outputs
+            when: "{{inputs.parameters.is-integration-test}} == false"
+          - name: run-integration-tests-loop
```
Comment (Bobgy) [inline test/sample_test.yaml]: nit: integration test and e2e test reads similar, I'm not sure people can easily understand how they are different.
If you cannot find a good name, we can leave the decision to later too, not blocking

**Comment by numerology** [comment]
> Confirm, when this PR is merged sample-test will no longer run the extra heavy samples, is that right?

That's right. I tend to separate the current sample test into sample test + integration test in one PR, so that there won't be a "blind time".

> Integration test and e2e test reads similar, I'm not sure people can easily understand how they are different.

Ack. To me, e2e test is something checks the functionality of a system end-to-end, but not necessarily calling external services/dependencies. Here it refers to test cases that can run **natively** within a GKE cluster. While integration test means tests that not only exercise the system end-to-end, but also invoke external services/dependencies. Here, it means that the test case will make RPC calls or request to services like AI platform/Dataflow/Dataproc, etc.

I will improve the documentation in a follow-up PR.

File: test/sample_test.yaml (hunk 1 above)
Comment (numerology) [reply test/sample_test.yaml]: Replied in the main thread.

**Comment by numerology** [comment]
@Bobgy Fixed a travis issue. PTAL thanks!

File: test/sample_test.yaml [hunk 2]
```diff
@@ -63,29 +66,46 @@ spec:
...
+                  value: "{{item}}"
+            withItems:
+              - xgboost_training_cm
+              - dataflow
               - ai_platform
+              - kubeflow_tf_serving
+              - container_build
               - parameterized_tfx_oss
               - iris
+            when: "{{inputs.parameters.is-integration-test}} == true"
```
Comment (Bobgy) [inline test/sample_test.yaml]: Shall we comment this line first, to make sure both tests are run in presubmit?

Comment (numerology) [reply test/sample_test.yaml]: Done.

**Pull request merged**